rsocket-core/src/main/java/io/rsocket/internal/ClientServerInputMultiplexer.java

```diff
@@ -74,7 +74,7 @@ public ClientServerInputMultiplexer(DuplexConnection source, PluginRegistry plug
                 if (frame.getType() == FrameType.SETUP) {
                   type = Type.STREAM_ZERO;
                 } else {
-                  type = Type.CLIENT;
+                  type = isServer ? Type.CLIENT : Type.SERVER;
```

Comment:
I don't think this change is sufficient - previously Client Requester also handled Keep-Alive frames, and now Connection gets closed eventually because of missing Keep-Alive acks.

Reply:
That's a good point -- I think what really needs to be done is separating stream zero into a DuplexConnection that gets handled independently of the client and server rather than trying to mix stream zero frames in with one or the other. That's a bit more work, but I had started some changes in that direction so probably time to dust that off and solve this properly.

Reply:
I came to this same conclusion working on the `DuplexConnection` straw man today.  It’s just so different (almost completely disjoint set of frame types, can’t be fragmented, special lifecycle, etc.) that it deserves to be treated differently.

Reply:
Indeed -- also the metadataPush method really doesn't belong in the RSocket interface, but in this stream zero handler, whatever we end up calling it.

Reply:
Yep, agreed.